File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/RoadRunner/messages/MecanumCommandMessage.java
package org.firstinspires.ftc.teamcode.RoadRunner.messages;

public final class MecanumCommandMessage {
    public long timestamp;
    public double voltage;
    public double leftFrontPower;
    public double leftBackPower;
    public double rightBackPower;
    public double rightFrontPower;

    public MecanumCommandMessage(double voltage, double leftFrontPower, double leftBackPower, double rightBackPower, double rightFrontPower) {
        this.timestamp = System.nanoTime();
        this.voltage = voltage;
        this.leftFrontPower = leftFrontPower;
        this.leftBackPower = leftBackPower;
        this.rightBackPower = rightBackPower;
        this.rightFrontPower = rightFrontPower;
    }
}

File: TeamCode/src/main/main/java/org/firstinspires/ftc/teamcode/drivetrain/messages/MecanumCommandMessage.kt
package org.firstinspires.ftc.teamcode.drivetrain.messages

class MecanumCommandMessage(
    voltage: Double,
    leftFrontPower: Double,
    leftBackPower: Double,
    rightBackPower: Double,
    rightFrontPower: Double
) {
    var timestamp: Long
    var voltage: Double
    var leftFrontPower: Double
    var leftBackPower: Double
    var rightBackPower: Double
    var rightFrontPower: Double

    init {
        this.timestamp = System.nanoTime()
        this.voltage = voltage
        this.leftFrontPower = leftFrontPower
        this.leftBackPower = leftBackPower
        this.rightBackPower = rightBackPower
        this.rightFrontPower = rightFrontPower
    }
}
